src/main/java/com/budgetapp/services/UserService.java

```diff
@@ -0,0 +1,35 @@
+package com.budgetapp.services;
+
+import com.budgetapp.entities.User;
+import com.budgetapp.repositories.UserRepository;
+import jakarta.persistence.EntityNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class UserService {
+
+    private final UserRepository userRepository;
+
+    public UserService(final UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public List<User> findAllUsers() {
+        return userRepository.findAll();
+    }
+
+    public User saveUser(final User user) {
+        return userRepository.save(user);
+    }
+
+    public User findUserById(final Long id) {
+        final User user = userRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(String.format("user with id : %s is not     found", id)));
```

Comment:
Just check your spacing here.
```suggestion
        final User user = userRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(String.format("user with id : %s is not found", id)));
```